app/mailers/favorite_mailer.rb

```diff
@@ -15,4 +15,14 @@ def new_comment(user, post, comment)
  # #19
      mail(to: user.email, subject: "New comment on #{post.title}")
    end
+
+   def new_post(post)
+    headers["Message-ID"] = "<posts/#{post.id}@your-app-name.example"
```

Comment:
The ideal way to set domain (your-app-name.example) is to use configuration variable. For example, 
in the application.rb

config.action_mailer.default_url_options = { host: "your domain name" }

And in the code, you can do 
Rails.configuration.action_mailer.default_url_options[:host] to get the domain name.